src/Altinn.Platform/Altinn.Platform.Authorization/Functions/Services/EventPusherService.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Altinn.Platform.Authorization.Functions.Clients;
+using Altinn.Platform.Authorization.Functions.Exceptions;
+using Altinn.Platform.Authorization.Functions.Models;
+using Altinn.Platform.Authorization.Functions.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+
+namespace Altinn.Platform.Authorization.Functions.Services
+{
+    public class EventPusherService : IEventPusherService
+    {
+        private readonly ILogger _logger;
+        private readonly BridgeClient _bridgeClient;
+
+        public EventPusherService(ILogger<EventPusherService> logger, BridgeClient bridgeClient)
+        {
+            _logger = logger;
+            _bridgeClient = bridgeClient;
+        }
+
+        public async Task PushEvents(DelegationChangeEventList delegationChangeEventList)
+        {
+            delegationChangeEventList.DelegationChangeEvents ??= new List<DelegationChangeEvent>();
+            try
+            {
+                if (_logger.IsEnabled(LogLevel.Debug))
+                {
+                    _logger.LogDebug(
+                        "Posting delegationChangeEventList numEventsSent={numEventsSent} changeIds={changeIds}",
+                        delegationChangeEventList.DelegationChangeEvents.Count,
+                        GetChangeIdsForLog(delegationChangeEventList));
+                }
+
+                HttpResponseMessage response =
+                    await _bridgeClient.PostDelegationEventsAsync(MapToBridgeModel(delegationChangeEventList));
+
+                if (!response.IsSuccessStatusCode)
+                {
+                    _logger.LogWarning(
+                        "Failed to post delegation events to bridge. resultCode={resultCode} reasonPhrase={reasonPhrase} resultBody={resultBody} numEventsSent={numEventsSent} changeIds={changeIds}",
```

Comment:
does {resultBody} actually match on property name of await response.Content.ReadAsStringAsync()
to replaces the correct variable? or is it just an order of the params thing?

how this mapping works is still a bit magical to me

Reply:
It's just parameter ordering. https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-6.0#log-message-template